src/session/session_prefetch.c

```diff
@@ -25,14 +25,26 @@ __wt_session_prefetch_check(WT_SESSION_IMPL *session, WT_REF *ref)
      * least one page from disk. The result of this function will subsequently be checked by cursor
      * logic to determine if pre-fetching will be performed.
      */
-    if (!F_ISSET(session, WT_SESSION_PREFETCH) || F_ISSET(session, WT_SESSION_INTERNAL) ||
-      F_ISSET(ref, WT_REF_FLAG_INTERNAL) || session->pf.prefetch_disk_read_count < 2) {
```

Comment:
Why did the `F_ISSET(ref, WT_REF_FLAG_INTERNAL)` check go?

There's something else confusing here - the previous code checked `disk_read_count < 2` and bailed, so the `prefetch_disk_one` stat could never get incremented. Am I reading this correctly?

Also, style is that we don't use braces for single-line blocks - the `disk_one` stat bump should have them removed.

Reply:
My bad, thanks for picking that one up. I'll add that check back in. Yes - I realised that previously the `prefetch_disk_one` stat could never get incremented which is why I've bumped that condition up.

Reply:
Thanks - and good spot!